tests/unit/types/arrays/test_neural_ops.py

```diff
@@ -52,13 +84,17 @@ def docarrays_for_embedding_distance_computation_sparse():
     return D1, D2
 
 
+@pytest.fixture
+def embeddings():
+    return np.array([[1, 0, 0], [2, 0, 0], [3, 0, 0]])
+
+
+@pytest.mark.parametrize('gen_doc_arrays', gen_docarray_combinations())
 @pytest.mark.parametrize(
     'limit, batch_size', [(1, None), (2, None), (None, None), (1, 1), (1, 2), (2, 1)]
 )
-def test_matching_retrieves_correct_number(
-    docarrays_for_embedding_distance_computation, limit, batch_size
-):
-    D1, D2 = docarrays_for_embedding_distance_computation
+def test_matching_retrieves_correct_number(gen_doc_arrays, limit, batch_size):
+    D1, D2 = gen_doc_arrays()
```

Comment:
I do not think this should be a callable, this should be the `docarrays` themselves